### DISCUSSION TITLE
Need help with my wallpaper widget

### DISCUSSION THREAD

**Discussion by Xtr4F** [Aylur/ags#405] [Q&A] [answered]
New to ags and GTK stuff, need some advice. What I'm trying to do is a simple wallpaper widget so I can set media as full screen wallpaper. After a brief digging through the documentation, I came up with the solution that's in the picture below:
![2024-05-01-230418_hyprshot](https://github.com/Aylur/ags/assets/105001002/c5447e3d-c933-418a-9f0a-bd24333555b1)

This works great with pictures, but I want to be able to work with gifs or videos as well. I've found that there is a PixbufAnimation for this purposes, but haven't found a way to resize the animation to match my monitor size. I've searched a lot, but it seems like there is very little information about GTK on the internet, especially for newbies. And even though the project has documentation doesn't really help me, it seems like it's written for people already familiar with GTK. Really want to learn how to do ags stuff, but for now it's all so overwhelming....

Any help or advice to point me in the right direction would be very appreciated!

**Comment by Aylur** [comment, marked as answer]
I haven't yet tried to play video in gtk, but for a wallpaper I would suggest going for another project because gtk is not the most resourceful for this imo
If you still want to do this, your best bet is to look at some other application that displays video to at least know that its doable with pixbufs or if another library is needed
https://github.com/LGFae/swww for gifs
https://github.com/GhostNaN/mpvpaper for videos
